tests/gtest/aviftest_helpers.h

```diff
@@ -17,6 +17,50 @@
 namespace avif {
 namespace testutil {
 
+//------------------------------------------------------------------------------
+// Duplicated from internal.h
+// Used for debugging. Define AVIF_BREAK_ON_ERROR to catch the earliest failure
+// during encoding or decoding.
+#if defined(AVIF_BREAK_ON_ERROR)
+static inline void avifBreakOnError() {
+  // Same mechanism as OpenCV's error() function, or replace by a breakpoint.
+  int* p = NULL;
+  *p = 0;
+}
+#else
+#define avifBreakOnError()
+#endif
+
+// Used by stream related things.
+#define AVIF_CHECK(A)     \
+  do {                    \
+    if (!(A)) {           \
+      avifBreakOnError(); \
+      return AVIF_FALSE;  \
+    }                     \
+  } while (0)
+
+// Used instead of CHECK if needing to return a specific error on failure,
+// instead of AVIF_FALSE
+#define AVIF_CHECKERR(A, ERR) \
+  do {                        \
+    if (!(A)) {               \
+      avifBreakOnError();     \
+      return ERR;             \
+    }                         \
+  } while (0)
+
+// Forward any error to the caller now or continue execution.
+#define AVIF_CHECKRES(A)              \
+  do {                                \
+    const avifResult result__ = (A);  \
+    if (result__ != AVIF_RESULT_OK) { \
+      avifBreakOnError();             \
+      return result__;                \
+    }                                 \
+  } while (0)
+//------------------------------------------------------------------------------
```

Comment:
Nit: Move the macro definitions outside the avif namespace to make it clear that namespaces don't apply to macros.

Later, we should use a different prefix for these macros.

Reply:
Done